api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -93,11 +93,39 @@ public Schema(int schemaId, List<NestedField> columns, Map<String, Integer> alia
     this.schemaId = schemaId;
     this.struct = StructType.of(columns);
     this.aliasToId = aliases != null ? ImmutableBiMap.copyOf(aliases) : null;
+
+    // validate IdentifierField
+    if (identifierFieldIds != null) {
+      Map<Integer, Integer> idToParent = TypeUtil.indexParents(this.struct);
+      identifierFieldIds.forEach(id -> validateIdentifierField(id, this.lazyIdToField(), idToParent));
+    }
+
     this.identifierFieldIds = identifierFieldIds != null ? Ints.toArray(identifierFieldIds) : new int[0];
 
     lazyIdToName();
   }
 
+  protected static void validateIdentifierField(int fieldId, Map<Integer, Types.NestedField> idToField,
```

Comment:
Why is this `protected`? Shouldn't it be package-private instead? I don't think that we intend to support subclasses of `Schema`.

Reply:
my mistake, changed it to protected because its called from `SchemaUpdate` removed it